### PR TITLE
Refactor: Unexplode Backup() function, pass BackupRequest as argument

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Similarly to https://github.com/vitessio/vitess/pull/10578 and https://github.com/vitessio/vitess/pull/10671, this PR unexplodes `Backup()` functions to include a `BackupRequest` instead of specific request fields like `concurrency` and `allowPrimary`.

I have used both `vtctldatapb.BackupRequest` as well as `tabletmanagerdata.BackupRequest`. I hope I've used both in their appropriate context - kindly consider this while reviewing.

## Related Issue(s)

#10578 , #10671 

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtctl/grpcvtctldserver/testutil/test_tmclient.go
Comment (ajm188) [inline go/vt/vtctl/grpcvtctldserver/testutil/test_tmclient.go:43]: this is already imported (directly below) as `tabletmanagerdatapb`

File: go/vt/vtctl/grpcvtctldserver/server.go
Comment (ajm188) [inline go/vt/vtctl/grpcvtctldserver/server.go:396]: 👍 per "i hope i've used each in their appropriate context", i agree with this delineation

File: go/vt/vttablet/faketmclient/fake_client.go
Comment (ajm188) [inline go/vt/vttablet/faketmclient/fake_client.go:39]: already imported as `tabletmanagerdatapb`

File: go/vt/vttablet/grpctmclient/client.go
Comment (ajm188) [inline go/vt/vttablet/grpctmclient/client.go:44]: imports

File: go/vt/vttablet/tabletmanager/rpc_backup.go
Comment (ajm188) [inline go/vt/vttablet/tabletmanager/rpc_backup.go:30]: can you change this to match the other aliased proto imports? (should just append `pb`)

File: go/vt/vttablet/tmclient/rpc_client_api.go
Comment (ajm188) [inline go/vt/vttablet/tmclient/rpc_client_api.go:32]: imports

File: go/vt/vttablet/tmrpctest/test_tm_rpc.go
Comment (ajm188) [inline go/vt/vttablet/tmrpctest/test_tm_rpc.go:43]: imports

**Review by ajm188 — COMMENTED** [review]
lgtm, pending the import duplication/naming

**Comment by shlomi-noach** [comment]
@ajm188 fixed all duplicate imports. Nice catch!

File: go/vt/vtcombo/tablet_map.go
Comment (ajm188) [inline go/vt/vtcombo/tablet_map.go:34]: for consistency, this should probably be `vschemapb` (and with the other `/proto/` imports below) (sorry i missed this one last time!)

**Review by ajm188 — APPROVED** [review]
lgtm, with the other comment i left (feel free to not block on it if you don't want to!)

File: go/vt/vtcombo/tablet_map.go
Comment (shlomi-noach) [reply go/vt/vtcombo/tablet_map.go]: `vschemapb` it is

**Comment by shlomi-noach** [comment]
`Cluster (topo_connection_cache)` consistently [fails](https://github.com/vitessio/vitess/runs/7624410926?check_suite_focus=true) on what seems to be related to backups. I'm unsure yet how this PR affects the test, but apparently it does.

**Comment by shlomi-noach** [comment]
Still clueless on the failing test.

**Comment by shlomi-noach** [comment]
huh. After merging `main`, `topo_connection_cache` now passed

**Pull request merged**
